designs/containers/vnf-deployment/vnf-deployment.md

```diff
@@ -84,23 +130,202 @@ public class ApplianceSoftwareVersion extends BaseEntity {
+    private String region;
+
+    @Column(name = "project_name", nullable = **true**)
+    private String projectName;
+
+    @Column(name = "project_id", nullable = **true**)
+    private String projectId;
+
+    @Column(name = "management_network_name", nullable = **true**)
+    private String managementNetworkName;
+
+    @Column(name = "management_network_id", nullable = **true**)
+    private String managementNetworkId;
+
+    @Column(name = "inspection_network_name", nullable = **true**)
+    private String inspectionNetworkName;
+
+    @Column(name = "inspection_network_id", nullable = **true**)
+    private String inspectionNetworkId;
+	
+	@Column(name = "external_id", nullable = **true**)
+    private String externalId;                           // Stores the Kubernetes deployment id.
+	
+	@Column(name = "namespace", nullable = **true**)
+    private String namespace;                           // Stores the Kuberneted deployment namespace.
+	
+	@OneToMany(mappedBy = "deploymentSpec", fetch = FetchType.LAZY)
+    private Set<InspectionPort> inspectionPorts = new HashSet<InspectionPort>(); // The set of inspection ports associated with this DS.
+	// ...
+}
+```
+#### Inspection Ports
+In Kubernetes, containers may be deleted and re-created at any time, without the direct control of OSC. This means it is possible for a container whose port is being used for traffic redirection to be deleted and replaced by Kubernetes. Because of that, OSC needs to have a way to update inspection ports registereed with the SDN controller with a new virtual port. To enable this, OSC will persist a new entity `inspection port` which will be mapped to a DAI and also have a reference to the DS. OSC will use this entity to identify whether an inspection port under a DS is not assigned to a virtual port and conform this state with the SDN controller accordingly.  
+Also because of the volatile nature of pods/containers OSC will refrain from creating inspection ports prior to an appliance being bound to a security group. Similarly an inspection port will only be deleted when an appliance is unbound.  This will avoid a high number of orphan inspection ports on the SDN controller.  
+
+> Note: As an alternative to having this new entity we could also store the inspection port id from the SDN controller directly on the DAI table. However, because a DAI maps to a container, associated by names, and new ones will have new names we must delete the DAI when a container is deleted. We cannot, however, lose the inspection port id (in this case no longer associated with the deleted DAI) since we need it to update the SDN controller. Thus mapping this relationship on a separate entity becomes necessary. 
+
+```java
+@Entity
+@Table(name = "INSPECTION_PORT")
+public class InspectionPort extends BaseEntity {
+	@Column(name = "externalId")
+    private String externalId;  // The SDN controller Element.getElementId() returned by registerInspectionPort

```

Comment:
i would move out the following fields from the DAI into this entity. Maybe we can add a type field indicating ingress/egress
```
    @Column(name = "inspection_os_ingress_port_id")
    private String inspectionOsIngressPortId;
    @Column(name = "inspection_ingress_mac_address")
    private String inspectionIngressMacAddress;

    @Column(name = "inspection_os_egress_port_id")
    private String inspectionOsEgressPortId;
    @Column(name = "inspection_egress_mac_address")
    private String inspectionEgressMacAddress;
```